File: src/main/java/art/arcane/racbot/app/BotRuntimeComponents.java
package art.arcane.racbot.app;

import art.arcane.racbot.config.BotConfig;
import art.arcane.racbot.service.EmbedFactory;
import art.arcane.racbot.service.PermissionService;
import art.arcane.racbot.service.SetupService;

public record BotRuntimeComponents(
    BotConfig config,
    SetupService setupService,
    PermissionService permissionService,
    EmbedFactory embedFactory) {}

File: lib/src/model/bot_runtime_components.dart
import 'package:racbot_nyxx/src/config/bot_config.dart';
import 'package:racbot_nyxx/src/service/embed_factory.dart';
import 'package:racbot_nyxx/src/service/role_persistence_service.dart';
import 'package:racbot_nyxx/src/service/runner_role_sync_service.dart';

class BotRuntimeComponents {
  final BotConfig config;
  final EmbedFactory embedFactory;
  final RolePersistenceService rolePersistenceService;
  final RunnerRoleSyncService? runnerRoleSyncService;

  const BotRuntimeComponents({
    required this.config,
    required this.embedFactory,
    required this.rolePersistenceService,
    required this.runnerRoleSyncService,
  });
}
